Description: Don't assume time_t is same as long.
 Per Linus' decree, any new architecture is to be safe from the Y2K38 bug.
 This means, time_t will be at least 64 bit, while long might be 32.
 An example of such an architecture is x32.
Author: Adam Borowski <kilobyte@angband.pl>

--- packagekit-1.0.1.orig/backends/aptcc/apt-intf.cpp
+++ packagekit-1.0.1/backends/aptcc/apt-intf.cpp
@@ -890,7 +890,9 @@ void AptIntf::emitUpdateDetail(const pkg
                         GTimeVal dateTime = {0, 0};
                         gchar *date;
                         date = g_match_info_fetch_named(match_info, "date");
-                        g_warn_if_fail(RFC1123StrToTime(date, dateTime.tv_sec));
+                        time_t time;
+                        g_warn_if_fail(RFC1123StrToTime(date, time));
+                        dateTime.tv_sec = time;
                         g_free(date);
 
                         issued = g_time_val_to_iso8601(&dateTime);
